### DISCUSSION TITLE
how to validade birth date for child and adult?

### DISCUSSION THREAD

**Discussion by julio3266** [react-hook-form/react-hook-form#8631] [General]
how do I define a min and max date for TextInput using type date? I'm trying using min and max, but is not validate ;x

<img width="734" alt="image" src="https://user-images.githubusercontent.com/52897217/177233374-969f839d-da6e-452f-a57a-c9bbf0926495.png">

**Comment by bluebill1049** [comment]
probably want to use validate function

**Reply by julio3266** [reply]
validate?

**Reply by julio3266** [reply]
do you have a example @bluebill1049 ?

**Reply by bluebill1049** [reply]
I don't have an example `validate: () => getValues()` as getValues will get you the entire form value and you can validate against that.

**Reply by julio3266** [reply]
I didn't quite understand how the validity would enter in this context

**Comment by Moshyfawn** [comment]
`min` and `max` attributes are for numbers only. As RHF is HTML validation standards compliant, it's `min`, `max` props also only support numbers.

You can utilize the `validate` callback _(like @bluebill1049 has already mentioned)_ to check if the input's value is within the valid date range like so _(the example uses [the date-fns library](https://date-fns.org/v2.28.0/docs/isWithinInterval) to work with dates for simplicity)_:
```js
validate: (date) =>
  isWithinInterval(date, { start: minDate, end: maxDate }) ||
  `The correct date is within ${format(
    minDate,
    dateFormat
  )} and ${format(maxDate, dateFormat)} interval`
```

Here's a minimal codesandbox to check out a working example: [the codesandbox example](https://codesandbox.io/s/blissful-sinoussi-k33lx4?file=%2Fsrc%2FApp.tsx)

**Reply by julio3266** [reply]
@Moshyfawn the problem is that I'm not using register, the structure here is a little different since I'm using useFieldArray and the defaultValue is being set in the interface using useForm, without the formValue
<img width="639" alt="image" src="https://user-images.githubusercontent.com/52897217/177344470-f437aec4-c787-45e2-8d0f-c6e83734d625.png">

**Reply by Moshyfawn** [reply]
What keeps you from providing the `validate` callback to your `Controller` / `useController`, if you are not using `register`?
```js
<Controller rules={{ validate: date => validateDate(date) }} />
```
`useFieldArray` doesn't handle your field values on it's own. You still register your inputs with either `register` or `Controller` / `useController` API. Both of these APIs support the `validate` callback and you can apply the same in your use-case, unless I'm missing something else.

**Reply by julio3266** [reply]
@Moshyfawn how do i keep the validation when i correct the values ​​entered in the input? I noticed that when I entered the validation it started to stop updating the hasError, and it keeps displaying the error continuously

**Reply by bluebill1049** [reply]
I think it's best if you can share a simple codsandbox point to the issue, it would be easier to see the uasge.

**Reply by Moshyfawn** [reply]
Man, it looks like you have some additional out of context logic in your example, that might possibly change the way to approach the validation. So, I'd suggest to make a **minimal** reproduction codesandbox with your use-case, because I can't go around reconstructing people's reproductions by their code snippets and screenshots.

That said, the example I've shown you is the way to validate a date range. I something's not working, the most possible cause if it can be something you do in your form else where. So, I can't do anything without a reproduction example

**Reply by julio3266** [reply]
@Moshyfawn I will make a codesandbox

**Reply by julio3266** [reply]
@Moshyfawn @bluebill1049  the[ sandobox link](https://codesandbox.io/s/react-hook-form-js-forked-gbsf9?file=/src/App.js)
